src/Util/ClassSourceManipulator.php

```diff
@@ -900,6 +925,31 @@ public function addUseStatementIfNecessary(string $class): string
         return $shortClassName;
     }
 
+    /**
+     * Builds a PHPParser attribute node.
+     *
+     * @param ClassNameDetails $attributeClass The attribute class which should be used for the attribute. If the class has a substitute
+     *                                         prefix, the node is built using the prefix. E.g. #[ORM\Column()] Otherwise, the
+     *                                         short name is used. E.g. #[Column()]
+     * @param array            $options        the named arguments for the attribute ($key = argument name, $value = argument value)
+     */
+    public function buildAttributeNode(ClassNameDetails $attributeClass, array $options): Node\Attribute
```

Comment:
If I understand correctly, we're using `ClassNameDetails` here so that we can communicate a possible prefix - e.g. `ORM\`. If so, could we instead add that as a 3rd arg?

```suggestion
    public function buildAttributeNode(string $attributeClass, array $options, string $attributePrefix = null): Node\Attribute
```

Reply:
Yes, but also because we need access to `ClassNameDetails::getShortName()`. Otherwise we get a FQCN in the in the attribute.. e.g. `#[Doctrine\Some\Long\Path\MyAttribute(....)]`. 

I suppose we could do something like:
```php
public function buildAttributeNode(string $attributeClass, array $options, string $attributePrefix = null): Node\Attribute
{
    $details = new ClassNameDetails($attributeClass, '\\');
}
```

And I think I was first going with that, except that the attribute method is then making assumptions about the 2nd argument for the CSD constructor - Minor detail which may be irrelevant in the grand scheme of things..

Reply:
Wait we also have MakerBundles `Str::getShortClassName()` - https://github.com/symfony/maker-bundle/blob/64318ec4a7870904e54f70e76e29ac8bc20e9875/src/Str.php#L123

That would work even better then init'ing a ClassNameDetails object..